import os
from PySide2 import QtCore, QtWidgets, QtGui

style = __file__.split('\\')[:-4]
style = os.path.join('/'.join(style), 'style.qss')


class Ui_Dialog(object):
    def setupUi(self, Dialog):
        with open(style, "r") as fh:
            Dialog.setStyleSheet(fh.read())

        path = __file__.split('\\')[:-4]
        path = "\\".join(path)

        sizePolicy = QtWidgets.QSizePolicy(QtWidgets.QSizePolicy.Fixed, QtWidgets.QSizePolicy.Fixed)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(Dialog.sizePolicy().hasHeightForWidth())
        Dialog.setSizePolicy(sizePolicy)
        Dialog.setMinimumSize(QtCore.QSize(300, 0))

        self.mainLayout = QtWidgets.QVBoxLayout(Dialog)
        self.mainLayout.setContentsMargins(10, 10, 10, 10)
        self.mainLayout.setSpacing(0)
        self.mainLayout.setObjectName("mainLayout")

        # LAYOUT
        self.FORMTAS = QtWidgets.QLabel(Dialog)
        sizePolicy = QtWidgets.QSizePolicy(QtWidgets.QSizePolicy.Fixed, QtWidgets.QSizePolicy.Fixed)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(self.FORMTAS.sizePolicy().hasHeightForWidth())
        self.FORMTAS.setSizePolicy(sizePolicy)
        self.FORMTAS.setObjectName("FORMTAS")
        self.mainLayout.addWidget(self.FORMTAS)
        self.listFormat = QtWidgets.QComboBox(Dialog)
        sizePolicy = QtWidgets.QSizePolicy(QtWidgets.QSizePolicy.Preferred, QtWidgets.QSizePolicy.Fixed)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(self.listFormat.sizePolicy().hasHeightForWidth())
        self.listFormat.setSizePolicy(sizePolicy)
        self.listFormat.setObjectName("listFormat")
        self.mainLayout.addWidget(self.listFormat)
        self.CAMERAS = QtWidgets.QLabel(Dialog)
        sizePolicy = QtWidgets.QSizePolicy(QtWidgets.QSizePolicy.Fixed, QtWidgets.QSizePolicy.Fixed)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(self.CAMERAS.sizePolicy().hasHeightForWidth())
        self.CAMERAS.setSizePolicy(sizePolicy)
        self.CAMERAS.setObjectName("CAMERAS")
        self.mainLayout.addWidget(self.CAMERAS)
        self.listCameras = QtWidgets.QComboBox(Dialog)
        self.listCameras.setObjectName("listCameras")
        self.mainLayout.addWidget(self.listCameras)
        self.FRAMERANGE = QtWidgets.QLabel(Dialog)
        sizePolicy = QtWidgets.QSizePolicy(QtWidgets.QSizePolicy.Fixed, QtWidgets.QSizePolicy.Fixed)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(self.FRAMERANGE.sizePolicy().hasHeightForWidth())
        self.FRAMERANGE.setSizePolicy(sizePolicy)
        self.FRAMERANGE.setObjectName("FRAMERANGE")
        self.mainLayout.addWidget(self.FRAMERANGE)
        self.framerange = QtWidgets.QWidget(Dialog)
        sizePolicy = QtWidgets.QSizePolicy(QtWidgets.QSizePolicy.Expanding, QtWidgets.QSizePolicy.Fixed)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(self.framerange.sizePolicy().hasHeightForWidth())
        self.framerange.setSizePolicy(sizePolicy)
        self.framerange.setObjectName("framerange")
        self.layoutFramgeRange = QtWidgets.QHBoxLayout(self.framerange)
        self.layoutFramgeRange.setContentsMargins(0, 0, 0, 0)
        self.layoutFramgeRange.setSpacing(8)
        self.layoutFramgeRange.setObjectName("layoutFramgeRange")
        self.frameRangeOptions = QtWidgets.QComboBox(self.framerange)
        sizePolicy = QtWidgets.QSizePolicy(QtWidgets.QSizePolicy.Expanding, QtWidgets.QSizePolicy.Fixed)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(self.frameRangeOptions.sizePolicy().hasHeightForWidth())
        self.frameRangeOptions.setSizePolicy(sizePolicy)
        self.frameRangeOptions.setObjectName("frameRangeOptions")
        self.layoutFramgeRange.addWidget(self.frameRangeOptions)
        self.frame_in = QtWidgets.QSpinBox(self.framerange)
        self.frame_in.setMaximumSize(QtCore.QSize(80, 16777215))
        self.frame_in.setButtonSymbols(QtWidgets.QAbstractSpinBox.NoButtons)
        self.frame_in.setMaximum(9999)
        self.frame_in.setObjectName("frame_in")
        self.layoutFramgeRange.addWidget(self.frame_in)
        self.frame_out = QtWidgets.QSpinBox(self.framerange)
        self.frame_out.setMinimumSize(QtCore.QSize(0, 0))
        self.frame_out.setMaximumSize(QtCore.QSize(80, 16777215))
        self.frame_out.setButtonSymbols(QtWidgets.QAbstractSpinBox.NoButtons)
        self.frame_out.setMaximum(9999)
        self.frame_out.setObjectName("frame_out")
        self.layoutFramgeRange.addWidget(self.frame_out)
        self.mainLayout.addWidget(self.framerange)
        # self.line = QtWidgets.QFrame(Dialog)
        # self.line.setFrameShape(QtWidgets.QFrame.HLine)
        # self.line.setFrameShadow(QtWidgets.QFrame.Sunken)
        # self.line.setObjectName("line")
        # self.mainLayout.addWidget(self.line)
        self.footer = QtWidgets.QWidget(Dialog)
        sizePolicy = QtWidgets.QSizePolicy(QtWidgets.QSizePolicy.Expanding, QtWidgets.QSizePolicy.Fixed)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(self.footer.sizePolicy().hasHeightForWidth())
        self.footer.setSizePolicy(sizePolicy)
        self.footer.setObjectName("footer")
        self.layoutButtons = QtWidgets.QHBoxLayout(self.footer)
        self.layoutButtons.setContentsMargins(0, 15, 0, 0)
        self.layoutButtons.setSpacing(8)
        self.layoutButtons.setObjectName("layoutButtons")
        self.buttonPreview = QtWidgets.QPushButton(self.footer)
        self.buttonPreview.setAutoDefault(False)
        self.buttonPreview.setObjectName("buttonPreview")
        self.layoutButtons.addWidget(self.buttonPreview)
        self.buttonRecored = QtWidgets.QPushButton(self.footer)
        self.buttonRecored.setAutoDefault(False)
        self.buttonRecored.setObjectName("buttonRecored")
        self.layoutButtons.addWidget(self.buttonRecored)
        self.mainLayout.addWidget(self.footer)

        self.retranslateUi(Dialog)
        QtCore.QMetaObject.connectSlotsByName(Dialog)
        # LAYOUT END

        widget = QtWidgets.QWidget()
        widget.setLayout(self.mainLayout)
        Dialog.layout().addWidget(widget)

    def retranslateUi(self, Dialog):
        _translate = QtCore.QCoreApplication.translate
        Dialog.setWindowTitle(_translate("Dialog", "ZS  Playblast"))
        self.FORMTAS.setText(_translate("Dialog", "FORMATS"))
        self.CAMERAS.setText(_translate("Dialog", "CAMERAS"))
        self.FRAMERANGE.setText(_translate("Dialog", "FRAME RANGE"))
        self.buttonPreview.setText(_translate("Dialog", "PREVIEW"))
        self.buttonRecored.setText(_translate("Dialog", "RECORD"))
